pkg/release/hook.go

```diff
@@ -76,6 +76,8 @@ type Hook struct {
 	Weight int `json:"weight,omitempty"`
 	// DeletePolicies are the policies that indicate when to delete the hook
 	DeletePolicies []HookDeletePolicy `json:"delete_policies,omitempty"`
+	// ExecutionDisabled is set to true when a hook has been filtered out and should not be executed
+	ExecutionDisabled bool `json:"-"`
```

Comment:
Given that this boolean is controlled by the filters, it's probably best to just make this a private variable instead. If I'm reading the code correctly, attempting to create a `release.Hook{ExecutionDisabled: true}` vs a `release.Hook{}` will not change its behaviour. In other words it's an internal state machine: it should not be exposed to the public API at this time.

Reply:
Since the action and this structure and in different packages, if I make it a private field, it will be inaccessible from the action. I can certainly do this, but I would need to make a public setter like `DisableExecution` or something similar to toggle the field. If that would be preferable, I am happy to make the change.

Another option would be to alter `execHook` rather than the hook structure. I briefly considered doing that but it initially seemed cleaner to mark the hook itself. Perhaps I was mistaken about that! I don't want to muddy the public API. I could do something like pass in a `activeHooks` variable to `execHook` and, if it is `nil`, default it to `rl.Hooks`. Then pass in the filtered hooks from `helm test` and `nil` from all of the other call locations.